### ISSUE TITLE
 Web3Task

### ISSUE THREAD

**Issue by viniblack** [onflow/developer-grants#247]
# Web3Task

## Grant category

- [x] Open source maintenance
- [ ] Developer tools / services
- [ ] Educational material

## Description

Web3Task is a platform that solves the problem of monetizing contributions by fragmenting tasks and requirements, inspired by the GitHub problem model.
Our goal is to tokenize the work of students and professionals by providing tangible proof of their value, tokenizing work by allowing students to gain experience while getting paid for their contributions to real Web 3.0 projects.

#### Target audience

- Students who want to enter the job market but encounter obstacles due to the predominance of freelancers, making it difficult for them to gain professional experience.
- Communities that want to hire professionals to perform specific tasks, but can't find them because there is no place to post their needs.

#### Evidence for the need

- There are currently more than [1.2 million technology students in Brazil](https://www.gov.br/inep/pt-br/assuntos/noticias/censo-da-educacao-superior/cursos-tecnologicos-crescem-11-5-em-um-ano#:~:text=about%20of%201%2C2%20milh%C3%A3o%20of%20students%20of%20this%20degree%20acad%C3%AAmico.), and most of them are unable to enter the job market because they lack professional experience.

### Proposed solution

It overcomes the significant barriers to entry that students face in the job market. We do this by tokenizing work and providing tangible evidence of value. We provide comprehensive community tools, including a scoring system to measure user outcomes, tokenization of work, portfolio registration, and a network to facilitate connections between users and projects.

### Impact

Flow communities will be able to quickly and easily find people to perform their specific tasks, while students will have the opportunity to build a solid career in Web 3 from the ground up by performing real-world project tasks.

## Milestones and funding
> **Note:**
> We have an MVP running on Ethereum and we're thinking of running the project through [EVM on Flow](https://forum.flow.com/t/evm-on-flow-beyond-solidity/5260) (Milestone 1).
> But if it's interesting for Flow, we can implement it directly on your blockchain using Cadence (Milestone 2).

### Milestone 1: Improving the MVP

| Milestone | Deliverables   | Timeline | Risks                   | USD proposal |
| --------- | -------------- | -------- | ----------------------- | -------------- |
| Develop a more user-friendly frontend | We already have the [frontend MVP](https://github.com/w3b3d3v/web3task-front) integrated with smart contracts on Ethereum [running on the web](web3task-front.vercel.app). We are working on designing a new, more user-friendly frontend with a more elegant UX. | 12 weeks   | - | 3,000 USD|
| Implement a backend with blockchain data   |With scalability in mind, we want to implement the [subgraph](https://github.com/w3b3d3v/web3task-front/pull/154) and create a backend to make on-chain queries cheaper.| 12 weeks| - | 5,000 USD|
| Marketing |We already have some potential users who are students from [partner communities](https://pt.w3d.community/ecossistema#parcerias). But we want to create a marketing campaign to attract more projects and users to the platform. | - | - | 2,000 USD|

Total funding proposed Milestone 1: **10,000 USD**

### Milestone 2: Implement Web3Task on the Flow Blockchain Network

| Milestone | Deliverables   | Timeline | Risks                   | USD proposal |
| --------- | -------------- | -------- | ----------------------- | -------------- |
| Implementing the MVP in Cadence  | [We have an Ethereum MVP](https://github.com/w3b3d3v/web3task-contracts) with the following features: Create Tasks, Start Tasks, Review Tasks, Complete Tasks, Cancel Tasks, Edit Task Titles, Edit Task Descriptions, Edit Task Deadlines, Edit Task Metadata, and manage access control for Set Role and Set Operator users. Initially, we plan to implement all of these Ethereum features in the Flow blockchain.  | 20 weeks  | Complexity of implementing what we have in Solidity for Cadence.  | 7,000 USD|
| Integrate Cadance contracts into the frontend | Integrate all the smart contracts developed in the "Implementing the MVP in Cadence" milestone into our [frontend](https://github.com/w3b3d3v/web3task-front). | 16 weeks   | Implement the libraries integrating smart contracts in Cadence with the frontend. | 3,000 USD|

Total funding proposed Milestone 2: **10,000 USD**

## Team

| Name | Role                | Bio | Contact         |
| ---- | ------------------- | --- | --------------- |
| Guilherme Neves | Founder and Product Manager  | Co-founder and Head of Blockchain at Blockful.io. He is a Degen very experienced in high leverage trading, DeFi, DAOs, NFTs & RWA. With over 20 projects delivered, he is a full stack developer specializing in smart contracts for EVM with extensive knowledge in auditing, consulting and general blockchain fundamentals. | [0xneves](https://www.linkedin.com/in/0xneves/) |
| Gustavo Boehm | Dev Smart Contract | A developer with five years of experience solving complex problems for large enterprises and new markets through application development. My passion for technology, cryptocurrencies and blockchain drives my daily work. In addition, I love challenges and spend my free time riding my bike and spending time with my family. I look forward to learning from you and contributing to your success. Make Money | [Guboehm](https://www.linkedin.com/in/gustavoboehm/) |
| Rafael Sanches | Dev Back-end  | Has worked in information security for 3 years, specializing in pentesting, QA and incident response. Notably, he gained international experience in Turkey for 2 months. Currently transitioning to blockchain with a focus on smart contract and backend development, he aims to leverage his security expertise to enhance his new career path. | [RafaDSan](https://www.linkedin.com/in/rafael-de-souza-sanches/) |
| Heron Lancellot | Dev Frontend | Student of Information Systems at UFF, passionate about blockchain and active in hackathons. Committed to purposeful development, I currently work as a Web3 frontend developer. My experience also extends to the backend, where I used Python, Flask and Django in technology initiation projects at the university. | [Heronlancellot](https://www.linkedin.com/in/heron-lancellot/) |
| Daniel Vianna | Dev Frontend  | Father, technology and business enthusiast. Dedicated to the development of web information systems for over 15 years. Since 2020 I have experienced the disruption that web3 protocols have caused in the markets. Currently working as a full-stack developer. | [Durdsvianna](https://www.linkedin.com/in/danielviannaweb3/) |
| Anna Bida | Quality Analysts  | With a degree in Business Administration and a passion for project and people management, I'm excited about the new forms of governance proposed by Web3, so I'm studying DAOs and helping this ecosystem by being the web3dev community leader. | [Annabida](https://www.linkedin.com/in/annaweb3/) |
| Andressa Cabral | Designer UX  | Application design and branding. Exploring the world of Web3 and following crypto market trends. I enjoy keeping up with the latest updates in AI technology and VR and AR experiences. | [Andressa-cabral](https://www.linkedin.com/in/andressa-cabral-19095b192/) |
| Matheus Iotti | Head of Communication  | Communications and Digital Technologies student with a passion for teaching and learning. I seek a holistic professional development that integrates different areas of knowledge for the development of digital products. Currently, I am studying Literature at the Federal University of Santa Catarina and pursuing personal studies in investments, administration and systems programming. | [Mitish98](https://www.linkedin.com/in/matheus-iotti/) |
| Vinicius Santana | Marketing and Community | Full-Stack Developer has been working with development for over 3 years, with experience in QA, frontend and back-end development. He has been working with the community for over a year and is committed to connecting with the web3 communities in Brazil and helping to create a more welcoming and easier ecosystem for beginners. | [Viniblack](https://www.linkedin.com/in/viniblack) |

**Comment by diegofornalha** [comment]
It's crucial to develop a specific strategy focused on Flow and Cadence, capitalizing on the uniqueness and impact of the large projects already.

I noticed a lack of detailed explanation on how students will transition from theoretical learning of the Cadence language to practical application in real tasks.

The use of the 2,000 dollars allocated for marketing also remains unclear. I suggest reconsidering this part of the plan, perhaps deferring it and initially focusing on an organic approach that involves participation in other established educational projects on Flow, such as the Emerald Academy.

In summary, the proposal needs a more precise alignment with Flow, a better-developed marketing strategy if included, and effective mechanisms to ensure a comprehensive integration process for students, from onboarding to active participation in Web3Task.

**Comment by Aliserag** [comment]
We appreciate your proposal. After careful review we've decided not to proceed at this time. Please stay connected to our Twitter and working groups to stay on top of the latest developments in the ecosystem and new grant opportunities.
